Framework/Algorithms/src/ConvertUnitsUsingDetectorTable.cpp

```diff
@@ -185,12 +188,12 @@ MatrixWorkspace_sptr ConvertUnitsUsingDetectorTable::convertViaTOF(
 
     std::size_t wsid = i;
 
-    try {
+    if (spectrumInfo.hasUniqueDetector(i)) {
```

Comment:
Is there anything in the `try` block that throws if there is more than one detectors? Otherwise should this be `hasDetectors()`?